File: FantasyRadio/src/main/java/ru/sigil/log/LogManager.java
package ru.sigil.log;

import java.util.Collection;
import java.util.LinkedList;

/**
 * User: Nikita
 * Date: 21.03.12
 */
public class LogManager {

    private static final String AppTAG = "FantasyRadio";
    private static final String TAG = LogManager.class.getSimpleName();
    private static final Collection<ICustomLogger> loggers = new LinkedList<>();

    public static boolean isDebug = true;


    private LogManager() {
    }

    private static void handle(final ICustomLogger.LogLevel logLevel, final String tag, final String component, final String message, final Exception e) {
        synchronized (loggers) {
            if (isDebug) {
                for (final ICustomLogger logger : loggers) {
                    logger.handle(logLevel, tag, component, message, e);
                }
            }
        }
    }

    public static void addLogger(final ICustomLogger logger) {
        synchronized (loggers) {
            if (null == logger) {
                e(TAG, "CmdLogHandler cannot be null");
            } else {
                loggers.add(logger);
            }
        }
    }

    public static void d(final String component, final String message) {
        handle(ICustomLogger.LogLevel.DEBUG, AppTAG, component, message, null);
    }

    public static void d(final String component, final String message, final Object... args) {
        handle(ICustomLogger.LogLevel.DEBUG, AppTAG, component, String.format(message, args), null);
    }

    public static void d(final String component, final String message, Exception e) {
        handle(ICustomLogger.LogLevel.DEBUG, AppTAG, component, message, e);
    }

    public static void e(final String component, final String message, final Exception e) {
        handle(ICustomLogger.LogLevel.ERROR, AppTAG, component, message, e);
    }

    public static void e(final String component, final String message) {
        handle(ICustomLogger.LogLevel.ERROR, AppTAG, component, message, null);
    }

    public static void w(final String component, final String message) {
        handle(ICustomLogger.LogLevel.WARNING, AppTAG, component, message, null);
    }

    public static void w(final String component, final String message, final Object... args) {
        handle(ICustomLogger.LogLevel.WARNING, AppTAG, component, String.format(message, args), null);
    }
}

File: FantasyRadio/src/main/java/ru/sigil/log/LogManager.kt
package ru.sigil.log

import java.util.*

private const val AppTAG = "FantasyRadio"

/**
 * Created by namelessone
 * on 09.12.18.
 */
object LogManager {
    private val TAG = LogManager::class.java.simpleName
    private val loggers = LinkedList<ICustomLogger>()

    private var isDebug = true

    private fun handle(logLevel: LogLevel, tag: String, component: String, message: String, e: Exception?) {
        synchronized(loggers) {
            if (isDebug) {
                for (logger in loggers) {
                    logger.handle(logLevel, tag, component, message, e!!)
                }
            }
        }
    }

    fun addLogger(logger: ICustomLogger?) {
        synchronized(loggers) {
            if (null == logger) {
                e(TAG, "CmdLogHandler cannot be null")
            } else {
                loggers.add(logger)
            }
        }
    }

    fun d(component: String, message: String) {
        handle(LogLevel.DEBUG, AppTAG, component, message, null)
    }

    fun d(component: String, message: String, vararg args: Any) {
        handle(LogLevel.DEBUG, AppTAG, component, String.format(message, *args), null)
    }

    fun d(component: String, message: String, e: Exception) {
        handle(LogLevel.DEBUG, AppTAG, component, message, e)
    }

    fun e(component: String, message: String, e: Exception) {
        handle(LogLevel.ERROR, AppTAG, component, message, e)
    }

    fun e(component: String, message: String) {
        handle(LogLevel.ERROR, AppTAG, component, message, null)
    }

    fun w(component: String, message: String) {
        handle(LogLevel.WARNING, AppTAG, component, message, null)
    }

    fun w(component: String, message: String, vararg args: Any) {
        handle(LogLevel.WARNING, AppTAG, component, String.format(message, *args), null)
    }
}